src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let conditionFunction = compareFunction;
+
+    if (!conditionFunction) {
+      conditionFunction = function (a, b) {
+        return String(a) > String(b) ? 1 : -1;
+      };
+    }
```

Comment:
This is a code quality issue. You are declaring `conditionFunction` twice: first as `compareFunction` and then as a new function if `compareFunction` is not provided. It would be cleaner to check if `compareFunction` is provided and if not, then assign the default function to `conditionFunction`. This way, you only need to declare `conditionFunction` once.